torch/_torch_docs.py

```diff
@@ -4530,6 +4530,11 @@ def merge_dicts(*dicts):
 
 Returns the median value of all elements in the :attr:`input` tensor.
 
+.. note::
+    For even-sized inputs, the smaller of the two middle elements in the sorted
+    order is returned. If the mean of the two middle elements is required, use
+    :func:`torch.quantile` with ``q=0.5`` instead.
+
 .. warning::
     This function produces deterministic (sub)gradients unlike ``median(dim=0)``
```

Comment:
Aside: it's very surprising to me that this version of median produces deterministic gradients but the other version, which actually computes indices, doesn't. We should probably fix "median_with_indices" to consistently return the FIRST valid median, just like we fixed argmin and argmax.

Reply:
I think, as you mention, that the other function returns non-deterministic indices and thus non-deterministic subgradients.
This one evenly distribute the gradient to all the inputs with the value used. So it always is deterministic.

Reply:
The reason it returns non-deterministic indices is because it uses quickselect to partition which is not stable. One way to solve it is to use 3-way quickselect that partitions input into <, = and >. Then choose the smallest index from the = segment.